integrationtests/mayavi/common.py

```diff
@@ -383,12 +386,18 @@ def run_standalone(self):
             type, value, tb = self.exception_info
             raise type, value, tb
 
+
     def run(self):
         """This starts everything up and runs the test.  Call main to
         run the test."""
         # Calls the users test code.
         try:
-            self.do()
+            if self.profile:
+                memory_assistant = MemoryAssistant()
+                memory_assistant.assertReturnsMemory(self.do_profile,
```

Comment:
As a design idea there is no reason to have a separate method for the `do_profile` if the user want to run the integration test and check for reference cycles then just run the normal test `do` using the memory_assistant.